tablecodec/tablecodec.go

```diff
@@ -388,10 +388,12 @@ func Unflatten(datum types.Datum, ft *types.FieldType, inIndex bool) (types.Datu
 	case mysql.TypeFloat:
 		datum.SetFloat32(float32(datum.GetFloat64()))
 		return datum, nil
+	case mysql.TypeVarchar, mysql.TypeString:
+		datum.SetString(datum.GetString())
```

Comment:
Many places of the existing code don't handle it correctly. Maybe we should just remove `KindString`, only use `KindBytes` in the future.